src/components/Register/Register.js

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from 'react';
+import { NativeModules } from 'react-native';
+import { Text, Button, Box, FormControl, ScrollView } from 'native-base';
+import { useToggle } from '../../hooks/useToggle';
+import { Username } from './Username';
+import { KBA } from './KBA';
+import { Password } from './Password';
+import { TextField } from './TextField';
+import { Specials } from './Specials';
+import { registrationTypeFactory } from './typeFactory.js';
+
+const { ForgeRockModule } = NativeModules;
+
+const callbackTypeToComponent = {
+  ValidatedCreateUsernameCallback: ({ label, setter }) => (
+    <Username label={label} setUsername={setter} key={label} />
+  ),
+  StringAttributeInputCallback: ({ label, setter, val }) => (
+    <TextField label={label} setter={setter} key={label} val={val} />
+  ),
+  BooleanAttributeInputCallback: ({ label, setter, val }) => (
+    <Specials label={label} setter={setter} val={val} key={label} />
+  ),
+  ValidatedCreatePasswordCallback: ({ label, setter, val }) => (
+    <Password label={label} setter={setter} val={val} key={label} />
+  ),
+  TermsAndConditionsCallback: ({ label, setter, val, terms }) => (
+    <Specials
+      label={label}
+      setter={setter}
+      val={val}
+      terms={terms}
+      key={label}
+    />
+  ),
+  KbaCreateCallback: ({ label, setter, val, questions }) => (
+    <KBA
+      label={label}
+      setter={setter}
+      val={val}
+      questions={questions}
+      key={label}
+    />
+  ),
+};
```

Comment:
My first instinct here is to want to see more of a `switch` pattern for component-callback mapping, rather than this kind of object map. It will allow us to have a one-to-many relationship, rather than one-to-one.